android/modules/ui/src/java/ti/modules/titanium/ui/widget/listview/ListItemProxy.java

```diff
@@ -301,6 +260,202 @@ public void call(Object data)
+			TiViewProxy proxy = bindsEntry.getValue();
+			if (proxy == this) {
+				continue;
+			}
+
+			// Do no continue if child proxy has no properties.
+			KrollDict proxyProperties = proxy.getProperties();
+			if (proxyProperties == null) {
+				continue;
+			}
+
+			// Fetch properties dictionary to store child view's changes to.
+			HashMap cachedProperties = null;
+			Object object = this.childProperties.get(bindsEntry.getKey());
+			if (object instanceof HashMap) {
+				cachedProperties = (HashMap) object;
+			} else {
+				cachedProperties = new KrollDict();
+				this.childProperties.put(bindsEntry.getKey(), cachedProperties);
+			}
+
+			// Copy all view proxy properties to this list item's dictionary.
+			// Saves view's current state when scrolled offscreen so we can restore it when scrolled back in.
+			for (KrollDict.Entry<String, Object> proxyPropertyEntry : proxyProperties.entrySet()) {
+				cachedProperties.put(proxyPropertyEntry.getKey(), proxyPropertyEntry.getValue());
+			}
+		}
+	}
+
+	private void copyChildPropertiesTo(TiViewProxy[] proxies, KrollDict template)
+	{
+		// Validate arguments.
+		if ((proxies == null) || (proxies.length <= 0) || (template == null)) {
+			return;
+		}
+
+		// Fetch child templates nested under given template.
+		final Object[] childTemplates = (Object[]) template.get(TiC.PROPERTY_CHILD_TEMPLATES);
+
+		// Update all child proxies.
+		for (int index = 0; index < proxies.length; index++) {
+			// Fetch child's template.
+			KrollDict childTemplate = null;
+			if ((childTemplates != null) && (index < childTemplates.length)) {
+				Object objectTemplate = childTemplates[index];
+				if (objectTemplate instanceof KrollDict) {
+					childTemplate = (KrollDict) objectTemplate;
+				} else if (objectTemplate instanceof HashMap) {
+					childTemplate = new KrollDict((HashMap) objectTemplate);
+				}
+			}
+
+			// Fetch child's assigned configuration.
+			TiViewProxy tiProxy = null;
+			String bindId = null;
+			KrollDict events = null;
+			if (childTemplate != null) {
+				tiProxy = (TiViewProxy) childTemplate.get("tiProxy");
+				bindId = childTemplate.getString(TiC.PROPERTY_BIND_ID);
+				events = childTemplate.getKrollDict(TiC.PROPERTY_EVENTS);
+			}
+
+			// Fetch child's properties.
+			TiViewProxy proxy = proxies[index];
+			final KrollDict properties = new KrollDict();
+			if (tiProxy != null) {
+				properties.putAll(tiProxy.getProperties());
+			}
+			final Object childPropertiesObj = this.childProperties.get(bindId);
+			if (childPropertiesObj instanceof HashMap) {
+				properties.putAll((HashMap) childPropertiesObj);
+			}
+
+			// Assign properties to child. For best performance, only update property if value is different.
+			for (KrollDict.Entry<String, Object> entry : properties.entrySet()) {
+				if (proxy.shouldFireChange(proxy.getProperty(entry.getKey()), entry.getValue())) {
+					proxy.setPropertyAndFire(entry.getKey(), entry.getValue());
+				}
+			}
```

Comment:
Since `setPropertyAndFire()` calls `shouldFireChange()`, it's doing the check twice. Although, I doubt this gains anything significant
```Java
for (KrollDict.Entry<String, Object> entry : properties.entrySet()) {
    final String key = entry.getKey();
    final Object newValue = entry.getValue();
    final Object oldValue = proxy.getProperty(key);
    
    if (proxy.shouldFireChange(oldValue, newValue)) {
        proxy.setProperty(key, newValue);
        proxy.firePropertyChanged(key, oldValue, newValue);
    }
}
```
I'm curious if you can get away with not updating the V8 object?
```Java
proxy.getProperties().put(key, newValue);
```

Reply:
I think we do need to let it update the V8 object's properties since an event fired for this child proxy will be referenced via the event's "source" property. The [shouldFireChange()](https://github.com/appcelerator/titanium_mobile/blob/e2894aa26b0dbae2cc86436ec891dc756264c99e/android/titanium/src/java/org/appcelerator/kroll/KrollProxy.java#L1039-L1049) method is pretty simple so I think it's fine.

**_Side Note**: Only updating properties that have changed (versus update all properties) is a huge performance improvement._